src/pages/Transaction/utils.tsx

```diff
@@ -227,24 +231,68 @@ export function useTransactionBalanceChanges(txn_version: string) {
     {variables: {txn_version}},
   );
 
+  function convertAddress(a: FungibleAssetActivity) {
+    return a.type.includes("GasFeeEvent")
+      ? a.gas_fee_payer_address ?? a.owner_address
+      : a.owner_address;
+  }
+
+  function convertType(activity: FungibleAssetActivity) {
+    if (activity.type.includes("GasFee")) {
+      return "Gas Fee";
+    }
+    if (activity.type.includes("Withdraw")) {
+      return "Withdraw";
+    }
+    if (activity.type.includes("Deposit")) {
+      return "Deposit";
+    }
+    if (activity.type.includes("StorageRefund")) {
+      return "Storage Refund";
+    }
+
+    return "Unknown";
+  }
+
+  function convertAmount(activity: FungibleAssetActivity) {
+    if (activity.type.includes("GasFeeEvent")) {
+      return -BigInt(activity.amount);
+    }
+    if (activity.type.includes("Withdraw")) {
+      return BigInt(-activity.amount);
+    }
+    return BigInt(activity.amount);
+  }
+
   const balanceChanges: BalanceChange[] =
     data?.fungible_asset_activities
       .filter((a) => a.amount !== null)
       .map((a) => ({
-        address: a.type.includes("GasFeeEvent")
-          ? a.gas_fee_payer_address ?? a.owner_address
-          : a.owner_address,
-        amount:
-          a.type.includes("GasFeeEvent") || a.type.includes("Withdraw")
-            ? BigInt(-a.amount)
-            : BigInt(a.amount),
-        type: a.type,
+        address: convertAddress(a),
+        amount: convertAmount(a),
+        type: convertType(a),
         asset: {
           decimals: a.metadata?.decimals,
           symbol: a.metadata?.symbol,
         },
       })) ?? [];
 
+  // Find gas fee and add a storage refund event
+  const gasFeeEvent = data?.fungible_asset_activities.find((a) =>
```

Comment:
Is StorageRefund an event we are creating? Do we need to account for it in this check?

Reply:
It's not a real event, I made one up so the UI looks nice.

Gas fee event is also not a real one, but it's created on the indexer